### PR TITLE
KV cache support in default handler

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: engines/python/setup/djl_python/huggingface.py [hunk 1]
```diff
@@ -223,6 +223,8 @@ def inference(self, inputs):
                     return Output().error(
                         "In order to enable dynamic batching, all input batches must have the same parameters"
                     )
+            if "prompts" in input_map:
```
Comment (lanking520) [inline engines/python/setup/djl_python/huggingface.py:226]: let's give a clearer name, cached_prompts

File: engines/python/setup/djl_python/rolling_batch/scheduler_rolling_batch.py [hunk 2]
```diff
@@ -154,18 +160,33 @@ def _get_input_ids(self, input_texts):
...
+        do_sample = str(parameters.get('do_sample', self.search_config.sampling)).lower() == "true"
         return SearchConfig(
             max_new_tokens=parameters.get('max_new_tokens',
                                           self.search_config.max_new_seqlen),
             top_k=parameters.get('top_k', self.search_config.topk),
             penalty_alpha=parameters.get('penalty_alpha',
                                          self.search_config.alpha),
             num_beams=parameters.get('num_beams', self.search_config.beam),
-            do_sample=parameters.get('do_sample', self.search_config.sampling),
+            do_sample=do_sample,
```
Comment (frankfliu) [inline engines/python/setup/djl_python/rolling_batch/scheduler_rolling_batch.py:184]: Assign to itself?

Comment (sindhuvahinis) [reply engines/python/setup/djl_python/rolling_batch/scheduler_rolling_batch.py:184]: No, there is a varialble at line 176

File: engines/python/setup/djl_python/huggingface.py (hunk 1 above)
Comment (sindhuvahinis) [reply engines/python/setup/djl_python/huggingface.py]: done

**Review by frankfliu — APPROVED** [review]

**Pull request merged**